### ISSUE TITLE
space-after-type-colon doesn't handle static methods or optional arguments

### ISSUE THREAD

**Issue by zertosh** [gajus/eslint-plugin-flowtype#119]
The following reports that there must be a `:` after the type annotation:

```js
declare class atom$Point {
  static fromObject(object: atom$PointLike, copy:? boolean): atom$Point,
  //                                        ^ Error here
  constructor(row: number, column: number): void,
  row: number,
  // ^ Error here
  column: number,
  //    ^ Error here
}
```

* If you remove the space between `static fromObject`, the errors on `row` and `number` go away.
* If you remove the `?` in `copy:?`, the error on `copy` goes away.

**Comment by danharper** [comment]
I don't think it's handling static `ObjectTypeProperty`s with `FunctionTypeAnnotation` (which I think only appears in `Declare*`)

For `copy:? boolean`, that is a lint error. You can either have `copy?: boolean` or `copy: ?boolean`.

**Comment by danharper** [comment]
The `static` issue is a Babylon bug - all properties after a static one are getting marked as static (in `declare class` only). I've opened a PR to fix it: https://github.com/babel/babylon/pull/135

I'll keep this open until I can retry on an updated Babylon build.

**Comment by danharper** [comment]
Fixed in Babylon [v6.11.0](https://github.com/babel/babylon/releases/tag/v6.11.0).
